app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -71,65 +84,71 @@ export function createPPOMMiddleware(
       ) {
         // eslint-disable-next-line require-atomic-updates
         const securityAlertId = uuid();
+        let securityAlertResponse: SecurityAlertResponse = {
+          reason: BlockaidResultType.Loading,
+          result_type: BlockaidReason.inProgress,
+          securityAlertId,
+        };
 
         ppomController
           .usePPOM(async (ppom: PPOM) => {
-            try {
-              const normalizedRequest = normalizePPOMRequest(req);
-
-              const securityAlertResponse = await ppom.validateJsonRpc(
-                normalizedRequest,
-              );
-
-              securityAlertResponse.securityAlertId = securityAlertId;
-              return securityAlertResponse;
-            } catch (error: any) {
-              sentry?.captureException(error);
-              const errorObject = error as unknown as Error;
-              console.error('Error validating JSON RPC using PPOM: ', error);
-              const securityAlertResponse = {
-                result_type: BlockaidResultType.Errored,
-                reason: BlockaidReason.errored,
-                description: `${errorObject.name}: ${errorObject.message}`,
-              };
+            const normalizedRequest = normalizePPOMRequest(req);
+            securityAlertResponse = await ppom.validateJsonRpc(
+              normalizedRequest,
+            );
+            securityAlertResponse.securityAlertId = securityAlertId;
+          })
+          .catch((error: unknown) => {
+            sentry?.captureException(error);
+            console.error(
+              'Error validating JSON RPC using PPOM: ',
+              typeof error === 'object' || typeof error === 'string'
+                ? error
+                : JSON.stringify(error),
+            );
 
-              return securityAlertResponse;
-            }
+            securityAlertResponse = {
+              result_type: BlockaidResultType.Errored,
+              reason: BlockaidReason.errored,
+              description:
+                error instanceof Error
+                  ? `${error.name}: ${error.message}`
+                  : JSON.stringify(error),
+            };
```

Comment:
This is going to result in capturing errors that are not directly related with the ppom validation... `usePPOM` might throw an error that is completely unrelated with validating request. For example any of [these](https://github.com/MetaMask/ppom-validator/blob/main/src/ppom-controller.ts#L319-L329) can throw an unrelated error. And I might be wrong, but I am assuming that `'User has securityAlertsEnabled set to false'` should not be a blockaid error description and captured within the securityAlertResponse.

Nevertheless those errors are not being caught anywhere (usePPOM returns a promise and we are not awaiting for it in order to be able to catch it [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/ppom/ppom-middleware.ts#L125)). So I do agree that we need to either await for usePPOM to resolve or if we do not need to await for it then we need to add the chained catch. But we need to distinguish between errors being thrown within the validation callback and errors being thrown by usePPOM.